### PR TITLE
add Sphero SDK

### PR DESCRIPTION
**Pull request by ryanfortner**
to close https://github.com/Botspot/pi-apps/issues/1552

I have made a few improvements to @zShqdows scripts, but I haven't tested this yet.

### CODE REVIEW

**Comment by zShqdows** [comment]
Thanks a lot for improving this for me!

**Comment by zShqdows** [comment]
Looks great.

**Comment by zShqdows** [comment]
> to close #1552
> 
> I have made a few improvements to @zShqdows scripts, but I haven't tested this yet.

How would i download the new code to give it a try?

**Comment by ryanfortner** [comment]
> How would i download the new code to give it a try?

You can clone this branch: `git clone https://github.com/ryanfortner/pi-apps.git -b patch-46 pi-apps-test`

then move the files from that branch to the pi-apps app folder in your home folder.

**Comment by Botspot** [comment]
> How would i download the new code to give it a try?

You can paste this pull-request number into the Pi-Apps Importer.
2022-03-10-114403_1920x1080_scrot

**Comment by ryanfortner** [comment]
Oops. Forgot about that.

File: apps/Sphero SDK/description [hunk 1]
```diff
@@ -0,0 +1,17 @@
+(BEFORE INSTALLATION: ENABLE SERIAL PORT, THEN DISABLE SERIAL CONSOLE IN RASPBERRY PI CONFIG)
```
Comment (Botspot) [inline apps/Sphero SDK/description:1]: First line of the description will be the tooltip. I think this can be improved.

Comment (zShqdows) [reply apps/Sphero SDK/description:1]: Should I change it to make more sense on how to activate those two?

Comment (zShqdows) [reply apps/Sphero SDK/description:1]: Or, explain its purpose?

**Comment by ryanfortner** [comment]
@Botspot @zShqdows I made some changes, is it any better?

**Comment by zShqdows** [comment]
> @Botspot @zShqdows I made some changes, is it any better?

Looks perfect as of now, I will wait and see what BotSpot says.

**Comment by zShqdows** [comment]
> @Botspot @zShqdows I made some changes, is it any better?

I would say keep the line that says the required Python version, but if you think it's unnecessary, then leave it out.

**Comment by zShqdows** [comment]
But, I think this is ready @Botspot @ryanfortner

**Comment by ryanfortner** [comment]
> I would say keep the line that says the required Python version, but if you think it's unnecessary, then leave it out.

I took it out because both Debian Buster and Bullseye have high enough Python versions for it to work.

**Comment by ryanfortner** [comment]
> But, I think this is ready 

Looks good to me!

**Comment by zShqdows** [comment]
> > I would say keep the line that says the required Python version, but if you think it's unnecessary, then leave it out.
> 
> I took it out because both Debian Buster and Bullseye have high enough Python versions for it to work. 

Ok, makes sense then

**Comment by zShqdows** [comment]
> > But, I think this is ready 
> 
> Looks good to me!

@Botspot We are ready to go!

**Comment by zShqdows** [comment]
@Botspot Do you think it's ready?

**Comment by zShqdows** [comment]
@theofficialgman What do you think about this app? Would it be ready for Pi-Apps?

**Comment by theofficialgman** [comment]
> @theofficialgman What do you think about this app? Would it be ready for Pi-Apps?

I'm honestly not really a fan. It just seems really niche, since you need the sphereo rvr robot to do anything.

I could imagine something more mainstream like lego mindstorm (since has been very popular over the years) being added.

also just from a quick lookover of the repo, it doesn't look like these scripts need to be run as sudo

**Comment by theofficialgman** [comment]
sorry accidentally hit close

**Comment by Crilum** [comment]
@theofficialgman Well, I think for people who have the Sphero, it would be really useful to be able to open up Pi-Apps, and be like:
"Wow, I can just hit install and program my robot, *on my Raspberry Pi*!"

**Comment by zShqdows** [comment]
> > @theofficialgman What do you think about this app? Would it be ready for Pi-Apps?
> 
> I'm honestly not really a fan. It just seems really niche, since you need the sphereo rvr robot to do anything.
> 
> I could imagine something more mainstream like lego mindstorm (since has been very popular over the years) being added.
> 
> also just from a quick lookover of the repo, it doesn't look like these scripts need to be run as sudo

It needs to be ran as sudo, or else the setup.py script will not run.

File: apps/Sphero SDK/install [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+rm -rf ~/sphero-sdk-raspberrypi-python.git
+
+git_clone https://github.com/sphero-inc/sphero-sdk-raspberrypi-python.git || error "Failed to clone repository!"
+cd ~/sphero-sdk-raspberrypi-python
+
+sudo python3 setup.py install || error "Failed to run setup.py!"
+sudo ./first-time-setup.sh || error "Failed to run first time setup!"
```
Comment (Jim-Konish) [inline apps/Sphero SDK/install:9]: Hi, I'm the primary maintainer of the Sphero RVR SDK.  You don't need sudo for `first-time-setup.sh`, and there's no need to run setup.py at all (it's left over from an earlier effort and I should really remove it).

For reference, [here is the instructions page](https://sdk.sphero.com/docs/getting_started/raspberry_pi/raspberry_pi_setup_advanced) for installation.

Thanks for adding our SDK into pi-apps!

Comment (zShqdows) [reply apps/Sphero SDK/install:9]: I understand. Thanks for taking the time to reply!

**Comment by zShqdows** [comment]
@ryanfortner The owner of the SDK stated to remove the setup.py command since it isn't needed. I don't know how to do commits.

**Comment by zShqdows** [comment]
and to not run first-time-setup.sh as sudo.

**Comment by ryanfortner** [comment]
Thanks @Jim-Konish , I've made the requested changes.

This works on my end and I think ready for merge, @Botspot ?

**Comment by ryanfortner** [comment]
Putting this PR on hold until the new GUI pr is merged (it will include a programming category).

**Comment by zShqdows** [comment]
> Putting this PR on hold until the new GUI pr is merged (it will include a programming category).

Ok, when is the new GUI expected to be released?

**Comment by ryanfortner** [comment]
I don't think there is a release date but you can check for updates here: https://github.com/Botspot/pi-apps/pull/1580

**Comment by theofficialgman** [comment]
now that I look at this again, it seems like a similar scenario to: https://github.com/Botspot/pi-apps/issues/1753 (flutter SDK)

@Botspot do we really want to add these software development kits to pi-apps? I personally feel like if none of the pi-apps maintainers use them, then they shouldn't really be added, since it just puts extra maintenance burden and support on us.

**Comment by zShqdows** [comment]
> now that I look at this again, it seems like a similar scenario to: https://github.com/Botspot/pi-apps/issues/1753 (flutter SDK)
> 
> 
> 
> @Botspot do we really want to add these software development kits to pi-apps? I personally feel like if none of the pi-apps maintainers use them, then they shouldn't really be added, since it just puts extra maintenance burden and support on us.
> 
> 
> 
> > Putting this PR on hold until the new GUI pr is merged (it will include a programming category).
> 
> 
> 
> also there has not been any changes to the category system in the new gui. if they are still needed/desired, those will come separately 

What do you mean by maintenance burden?

**Comment by theofficialgman** [comment]
> What do you mean by maintenance burden?

as upstream projects change, sometimes buildscripts and install methods change or break. if there is not a maintainer/contributor at pi-apps to keep up with these changes, then you are left with either a broken app that users complain about, or we are forced to remove it.

as I mentioned, none of us are professional software developers (I think...) so we can't personally be the person to provide maintenance and support for Sphero SDK. sphero sdk is also unmaintained by the developers over there, so again, maintenance for their SDK would end up falling on pi-apps

**Comment by Jim-Konish** [comment]
It is being maintained, just very intermittently as a side project.

If distribution through pi-apps requires me to submit a pi-apps PR any time I do make an update, I'd rather have users just get it directly from the Sphero GitHub instead.

**Comment by Crilum** [comment]
@Jim-Konish If you distribute it though GitHub, we can make an automatic update script to update the Pi-Apps repo with the latest Release from GitHub, so you shouldn't have to do anything.

**Comment by theofficialgman** [comment]
> @Jim-Konish If you distribute it though GitHub, we can make an automatic update script to update the Pi-Apps repo with the latest Release from GitHub, so you shouldn't have to do anything.

you will need to do something like create a tag on the repo to indicate a new version has been released. it would not be ideal to update with every commit especially if the commits only include readme changes (like sen before)

**Comment by zShqdows** [comment]
I personally think that this can be added, as @Crilum stated, we **_can_** use an out update script, which would allow easy access to the latest repo.

**Comment by zShqdows** [comment]
Without having the need for users to manually update, because after a while, it would get frustrating.

**Comment by ryanfortner** [comment]
@Botspot would you be okay with merging this for now? I can and will make the necessary changes to the scripts if @Jim-Konish decides to start tagging releases in the future.

**Pull request merged**